util/proxy/scope.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package proxy
+
+import (
+	"crypto/tls"
+	"time"
+
+	"k8s.io/client-go/rest"
+)
+
+// Scope is a proxy scope
+type Scope struct {
```

Comment:
How do you determine when a field goes in or out of the Scope object? I am a little wary of structs named `Scope` as they tend to not have any clear boundaries and it's easy to stick everything inside of it. I don't really see a pattern in this Scope struct, but maybe you had one in mind? If so it would be good to document this or perhaps rename it to something less all-encompassing